src/fs/filesystem.ts

```diff
@@ -181,17 +183,25 @@ export class FileSystem {
 
   async ls(path: DirectoryPath): Promise<BaseLinks> {
     if (pathing.isFile(path)) throw new Error("`ls` only accepts directory paths")
-    return this.runOnTree(path, false, (tree, relPath) => {
-      return tree.ls(relPath)
+    return this.runOnNode(path, false, (node, relPath) => {
+      if (typeCheck.isFile(node)) {
```

Comment:
ah right... paths let us check the node type ahead of time (as long as everything is structured correctly). So wtih proper pathing, we should never hit this error

loving the new path primitives :ok_hand: